lib/misc.c

```diff
@@ -1282,3 +1282,35 @@ gftp_get_share_dir (void)
   return (gftp_share_dir);
 }
 
+char *
+gftp_format_file_size(off_t bytes) {
+  char tmp[20];
+  char ret[40];
+  char *pointzero = 0;
+  if (bytes >= 1099511627776) {
+    snprintf(tmp, sizeof(tmp), "%.2f", (float) bytes / 1099511627776);
```

Comment:
I believe that bytes can be treated as a float before the division by some compilers. I think this should be:

`(float) (bytes / (1024*1024*1024*1024))`

Reply:
Now I use `off_t TB = 1024LL * 1024LL * 1024LL * 1024LL;` see updated commit..

Reply:
I mean I changed (float) to (double)